protocol/bc/txhashes.go

```diff
@@ -4,14 +4,25 @@ type (
 	// TxHashes holds data needed for validation and state updates.
 	TxHashes struct {
 		ID Hash
+
 		// contains OutputIDs and retirement hashes.
-		// each OutputID is also the corresponding UnspentID
-		ResultHashes []Hash
-		Issuances    []struct {
+		Results   []ResultInfo
+		Issuances []struct {
 			ID           Hash
 			ExpirationMS uint64
 		}
-		VMContexts []*VMContext // one per old-style Input
+		SpentOutputIDs []Hash
```

Comment:
Do the entries in this list correspond 1:1 with the inputs of the oldtx? (And for inputs that aren't spends, are those entries the zero hash?) Please add a comment saying so.